src/features/farming/crops/components/Field.tsx

```diff
@@ -40,6 +43,36 @@ export const Field: React.FC<Props> = ({
   const clickedAt = useRef<number>(0);
   const field = game.context.state.fields[fieldIndex];
   const [showCropDetails, setShowCropDetails] = useState(false);
+  const [showRemoveWarning, setShowRemoveWarning] = useState(false);
+  const [lastCropRemovedTime, setLastCropRemovedTime] = useState<number | null>(
+    null
+  );
+
+  // Record when some crop was last removed
+  useEffect(() => {
```

Comment:
One concern of this approach is that the component becomes more dependent on external conditions and events. Previously most of the state was internalised or focussed around a single field, now other fields can affect each other.

We are beginning land expansion and adding more functionality every week. To help scale and avoid bugs/maintenance we want to try keep the `Field.tsx` more contained.

After playing around with the modal/removing crops I was wondering if we could simplify this a bit and bring it back to the 'tappy' nature of the game.

When a Rusty Shovel is equipped, we can employ a double tap to remove a crop. This would work similar to chopping a tree, but instead if would remove the crop. Going down this avenue would help with the consistence of user actions in the game (tapping to harvest/mine/remove a resource).

To assist with the 'danger' aspect, I would consider using the red progress bar.

![redbar_03](https://user-images.githubusercontent.com/11745561/178861339-1df021b3-aa4b-4832-b6cf-d3e4b140fe25.png)

Reply:
Got it! Can you just confirm my finalized understanding + answer me one question:

> In terms of equipping every player with a Rusty Shovel, I realized this may just be easier that we hardcode this on the API that every farm has the shovel. This will simplify the workflow of an 'airdrop'.

Perfect :+1: - just to confirm, I should not "put the rusty shovel next to the farm sign" - I should just leave the code as-is and have you hardcode the rusty shovel in the API. Correct?

> When a Rusty Shovel is equipped, we can employ a double tap to remove a crop.

Got it :+1: ! Just to be clear, this means we should kill the "warning modal" component altogether, correct?

> We will still want to go forward with the Basic shovel as well and release this at the same time even if it is a bit of scope creep.

Please just confirm me:

- Sell the "Basic Shovel" at the Tools shop on Human Side?
- AssetID 308 ? (rusty shovel is 307)
- Ingredients? Perhaps 5 stone, 10 wood, 3 SFL?

> It is extremely hard to take functionality away from users so we can't give this functionality (removing crops) with the initial Rusty shovel.

The only thing I'm unclear on is what is the functionality difference between the "Rusty Shovel" VS "Basic Shovel"? Are you saying the "Rusty Shovel" will simply sit in player's inventory without any functionality, and only the "Basic Shovel" will allow crop removal?

Reply:
1. Yes, you don't need to place the rusty shovel anywhere. We will sort that out
2. Yes, let's kill it 🔪 
3. Yes, Sell basic shovel at the tools shop
4. IDs are perfect
5. Let's make it more work for players 💪 10 iron, 20 wood
6. The rusty shovel will 'do nothing' - essentially it's functionality is that it lets you plant/harvest crops. When it gets stolen you can no longer farm